tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -204,48 +204,58 @@ Status StorageManager::array_open_for_reads(
     std::vector<FragmentMetadata*>* fragment_metadata) {
   STATS_START_TIMER(stats::Stats::TimerType::READ_ARRAY_OPEN)
 
-  // Open array without fragments async. This loads the array schema which
-  // is not needed in this function, so it can safely be loaded in parallel
-  // to listing of the fragment metadata and loading the consolidated fragment
-  // metadata file
-  auto open_array = (OpenArray*)nullptr;
-  std::vector<ThreadPool::Task> load_array_schema_task;
-  load_array_schema_task.emplace_back(io_tp_->execute([&, this]() {
-    RETURN_NOT_OK_ELSE(
-        array_open_without_fragments(array_uri, enc_key, &open_array),
-        *array_schema = nullptr);
-    return Status::Ok();
-  }));
-
-  // Determine which fragments to load
   std::vector<TimestampedURI> fragments_to_load;
```

Comment:
Several of these variables are modified in `load_array_fragments_task` on a different thread. Wondering if we should we put them in a local struct with a mutex just to flag that fact for future readers? (and `unique_lock` in the task)

Reply:
Seems ok as is to me. I don't see any reason to place them in a struct. If anything, we could make the lambda capture explicit on the variables instead of using `&`.

Reply:
Ok, updated with explicit capture and added a comment at the declaration site.